style: Fancier menus, more SIP-34-ish

rusackas opened the pull request:
### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
Moving the menus toward SIP-34. Design details are a bit up for grabs... I'll continue to steer with @steejay. Probably best not to merge until https://github.com/apache/incubator-superset/pull/10401 is merged, so this doesn't cause headaches alternating between the two menus.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
<!--- Skip this if not applicable -->
![looping](https://user-images.githubusercontent.com/812905/88503594-0fc00b00-cf87-11ea-8992-3380d5daa32f.gif)

### TEST PLAN
<!--- What steps should be taken to verify the changes -->

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [ ] Has associated issue:
- [x] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

rusackas committed:
style: shinier menus

rusackas committed:
fixing mouseover glitch

rusackas committed:
unused import

rusackas committed:
rm emotion-reset

rusackas committed:
restoring default config values

rusackas committed:
lint fixes :sparkles:

rusackas committed:
a bit more pizzaz to the underlines (max's idea), fading active background

etr2460 commented on superset-frontend/src/components/Menu/Menu.tsx:
```diff
@@ -56,32 +56,52 @@ export interface MenuProps {
       opacity: 0;
-      transition: opacity ${({ theme }) => theme.transitionTiming * 2}s;
+      transform: translateX(-50%);
+      transition: all ${({ theme }) => theme.transitionTiming}s !important;
```
yikes... why do we need an `!important` here?

rusackas replied:
Yeahhhhhhh, I have a hatred for `!important` too. This was just because it's overriding some deeeep selector stuff from Bootstrap. I can add the full selector stack instead, I suppose. Will look into it.

etr2460 replied:
I think it's worth seeing how messy it looks either way. It might be better to show the full selector stack regardless, since then it's more obvious to developers that this is fragile logic. `!important` can hide a bunch of nuance

etr2460 commented on superset-frontend/src/components/Menu/MenuObject.tsx:
```diff
@@ -44,17 +45,12 @@ export default function MenuObject({
-      &nbsp; {label}
-    </>
-  );
+  const navTitle = <>{label}</>;
```
this can simply be `const navTitle = label;`? Or we can remove `navTitle` entirely and use `label` instead

etr2460 commented on superset-frontend/src/components/NavDropdown/index.tsx:
```diff
@@ -0,0 +1,74 @@
+ */
+
+import styled from '@superset-ui/style';
+import { NavDropdown as RBNavDropdown } from 'react-bootstrap';
```
Can we name this `BootstrapNavDropdown`? Or `ReactBootstrapNavDropdown`. I agree the latter is a bit verbose, but `RB` was tough for me to grok at first

etr2460 commented on superset-frontend/src/components/NavDropdown/index.tsx:
```diff
@@ -0,0 +1,74 @@
+
+const NavDropdown = styled(RBNavDropdown)`
+  & > a.dropdown-toggle {
+    padding-right: ${({ theme }) => theme.gridUnit * 6}px !important;
```
again, a bit odd we need the `!important`

etr2460 commented on superset-frontend/src/components/NavDropdown/index.tsx:
```diff
@@ -0,0 +1,74 @@
+    transition: background ${({ theme }) => theme.transitionTiming}s;
+  }
+  &.open > a {
+    background: ${({ theme }) => theme.colors.primary.light4} !important;
```
and same important comment here

etr2460 reviewed (commented):
My biggest question is about the `!important`s. Is there a way to avoid these? And if not, can we add comments explicitly saying why we need to use them?

To be clear, I would much, much rather we get rid of them. But if it's impossible, then a comment would help remove a bit of the code smell

rusackas committed:
simplifying navtitle -> label

rusackas committed:
RBNavDropdown  -> ReactBootstrapNavDropdown

rusackas committed:
nixing whitespace

rusackas committed:
replacing !importants with better selector specificity

rusackas committed:
moving Menu LESS into Emotion

rusackas committed:
no more border!

rusackas committed:
fixing border issue

rusackas committed:
language picker, account dropdown now use new dropdown component

rusackas committed:
nixing whitespace in comment

rusackas committed:
nixing duplicate styling

rusackas committed:
removing borders on FAB navbar

rusackas committed:
explicit font coloring

rusackas committed:
linting

rusackas commented:
Got the two lil' menus on the right to conform, for good measure. 
![rightmenu](https://user-images.githubusercontent.com/812905/91127111-65590780-e65a-11ea-94da-76361dc3786b.gif)

rusackas commented:
Addressed PR comments and made lots of touchups. @mistercrunch / @etr2460 if you wouldn't mind giving it another glance, that'd make me feel better before merging. @riahk / @nytai might want to take a glance too.

rusackas commented:
Addresses aspects of #8976
